### PR TITLE
feat:[CDS-72798]: Codepulse Annotations

### PR DESCRIPTION
**Pull request by puthrayaharness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53760)

### CODE REVIEW

**Comment by puthrayaharness** [comment]
trigger uts

**Review by anilchowdhury-harness — DISMISSED** [review]

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 611 of 611 files at r1, 327 of 327 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53760)* status: 938 file(s) to review. Waiting on team(s): sto.

**Comment by puthrayaharness** [comment]
trigger uts review

**Review by archit-harness — DISMISSED** [review]

**Review by ankittiwari-harness — APPROVED** [review]

**Review by dhruv-harness — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53760)* status: 938 file(s) to review. Waiting on team(s): sto. 

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/VerificationApplication.java` line 294 at r2](https://reviewable.io/reviews/harness/harness-core/53760#-Nfz9gBhD_txtqRDSfm6:-Nfz9gBhD_txtqRDSfm7:beti3xs) ([raw file](https://github.com/harness/harness-core/blob/759a6c5273b373368befc074c628440e25182839/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/VerificationApplication.java#L294)):*
> ```Java
> import ru.vyarus.guice.validator.ValidationModule;
> 
> @CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_PIPELINE})
> ```

I am not sure if this can be tagged to CDS, this is shared by the SRM
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/VerificationConfiguration.java` line 80 at r2](https://reviewable.io/reviews/harness/harness-core/53760#-Nfz9oVy8DCGDtN4BQXn:-Nfz9oVy8DCGDtN4BQXo:b-xvgyit) ([raw file](https://github.com/harness/harness-core/blob/759a6c5273b373368befc074c628440e25182839/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/VerificationConfiguration.java#L80)):*
> ```Java
> import org.reflections.Reflections;
> 
> @CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_PIPELINE})
> ```

Same for this one, it is shared by SRM

File: idp-service/src/main/java/io/harness/idp/app/IdpApplication.java [hunk 1]
```diff
@@ -123,6 +125,8 @@
 /**
  * The main application - entry point for the entire Wings Application.
  */
+
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_PIPELINE})
```
Comment (sathish-soundarapandian) [inline idp-service/src/main/java/io/harness/idp/app/IdpApplication.java:129]: Is this just placeholder for now ? and later should be changed by us to relevant module and components ?
For IDP, ProductModule should be defined as IDP and HarnessModuleComponent will also change accordingly.

Comment (puthrayaharness) [reply idp-service/src/main/java/io/harness/idp/app/IdpApplication.java:129]: Yes, that's correct. We have auto-annotated based on recent commits made.

**Comment by puthrayaharness** [comment]
trigger all

**Comment by puthrayaharness** [comment]
trigger uts

**Review by arvind-choudhary-h — APPROVED** [review]

**Review by anilchowdhury-harness — APPROVED** [review]

**Comment by puthrayaharness** [comment]
trigger review

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 26 of 611 files at r1, 327 of 327 files at r2, 585 of 585 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53760)* status: 585 file(s) to review. 1. unresolved discussions. Waiting on team(s): sto.

**Pull request merged**